code.js

```diff
@@ -1,3 +1,85 @@
 function depthFirstSearch(graph, startNode, targetNode) {
+    
+    let isThere = false;
+    let stack = [];
+    let visited = [];
+    stack.push(startNode);
+    
+    while (stack.length > 0) {
+        //console.log("new loop");
+        visited.push(stack[stack.length - 1]);
+        if (stack[stack.length - 1] == targetNode) {
+            return visited;
+        }
+        
+        let tmp = stack[stack.length - 1];
+        stack.pop();
+        for (let i = 0; i < graph[tmp].length; i++) {
+            for (let j = 0; j < visited.length; j++) {
+                if (visited[j] == graph[tmp][i]) {
```

Comment:
Not sure what you're doing here. Is this just to avoid duplicates in `visited`?

Reply:
Yes, this is to go through the adjacent nodes of the one we are currently at and check if we have already visited it. If we have not visited it then it pushes it to the stack.